Client/Frontend/Browser/BrowserPrompts.swift

```diff
@@ -35,12 +42,33 @@ class JSPromptAlertController: UIAlertController {
 protocol JSAlertInfo {
     func alertController() -> JSPromptAlertController
     func cancel()
+    /// Informs JSPromptAlertController that the completion handler has already been called.
+    var didComplete: Bool { get }
 }
 
-struct MessageAlert: JSAlertInfo {
+class MessageAlert: JSAlertInfo {
     let message: String
     let frame: WKFrameInfo
-    let completionHandler: () -> Void
+    /// Do not access this directly. Use completionHandler()
+    private var _completionHandler: (() -> Void)?
```

Comment:
We tend to stay away from this older C-style of naming encapsulated data with `_` prefixes. Generally we prefer the more "modern" Swifty way of, first: evaluate why we need this to be private in the first place; and, second: if it does need to be private, then naming it something else that's related and still clear in its intent.

Also applies in the other places this appears in these changes.

Reply:
I gave it the _ to prevent future implementors of the code accessing the variable directly without paying attention to the requirement that it should only be accessed via the completionHandler() call. If you have a better name that would be great.

Reply:
The issue wasn't the intention, it was the implementation for our codebase style. haha. The intention is clear with the keyword `private`. I think a better solution might be calling the private variable `completionHandler` and the calling the function the verb of the action you're doing. Perhaps `callCompletionHandler` or `performCompletionHandler`. This links the two things, but also separates their intent.

Reply:
No - Private doesn't give the intent at all on its own in this case. As well as meaning "you can't use this from outside this class", the '_' is also there to try to ensure the variable is never used by any other code within the class that is outside of the completionHandler() function itself.